src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -531,3 +531,23 @@ def _validate_and_return_run_name(run: Union[str, "Run"]) -> str:
         elif isinstance(run, str):
             return run
         raise InvalidRunError(f"Invalid run {run!r}, expected 'str' or 'Run' object but got {type(run)!r}.")
+
+    def _generate_output_path(self) -> Path:
+        config = Configuration.get_instance()
+        path = config.get_run_output_path()
+        if path is None:
+            path = Path.home() / PROMPT_FLOW_DIR_NAME / ".runs"
+        else:
+            try:
+                path = Path(path.replace(FLOW_DIRECTORY_MACRO_IN_CONFIG, self.flow.resolve().as_posix()))
+                path = (path / ".runs").resolve()
+                path.mkdir(parents=True, exist_ok=True)
+            except Exception:  # pylint: disable=broad-except
```

Comment:
may there be execution in ready only file systems?

Reply:
there might be, but:

1. it's a user specified config, so he/she should be aware of this
2. there is one `mkdir` here, in read only file system, it will raise exception, caught, then fall back to default path